### PR TITLE
Implement upstream download code style

### PR DESCRIPTION
**Pull request by ChillerDragon**
In https://github.com/teeworlds/teeworlds/pull/2556 teeworlds introduced temporary download files for maps. DDNet already has that but a bit different. Those temporary files are now also deleted on error like it is done in the upstream. The main goal of this pr is to reduce code diff to upstream.

Tested official ddnet maps over http and other maps.

## Checklist

- [X] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by heinrich5991** [comment]
Can you describe what this changes functionally?

**Comment by ChillerDragon** [comment]
> Can you describe what this changes functionally?

It is mostly renaming and restructure. New is the deletion of the temporary file https://github.com/ddnet/ddnet/pull/3991/files#diff-9008c287b2e3181f671692c3dd859c3921d5ee178753b62cc254d9abd1b1e5a3R2373 if the download does not finish.

**Comment by def-** [comment]
Fine from my side. @heinrich5991 opinion?

File: src/engine/client/client.cpp [hunk 1]
```diff
@@ -1658,16 +1661,18 @@ void CClient::ProcessServerPacket(CNetChunk *pPacket)
...
+					}
 
-					str_format(m_aMapdownloadFilename, sizeof(m_aMapdownloadFilename), "downloadedmaps/%s", aTempFilename);
+					// start map download
+					FormatMapDownloadFilename(pMap, pMapSha256, MapCrc, false, m_aMapdownloadFilename, sizeof(m_aMapdownloadFilename));
+					FormatMapDownloadFilename(pMap, pMapSha256, MapCrc, true, m_aMapdownloadFilenameTemp, sizeof(m_aMapdownloadFilenameTemp));
 
 					char aBuf[256];
-					str_format(aBuf, sizeof(aBuf), "starting to download map to '%s'", m_aMapdownloadFilename);
+					str_format(aBuf, sizeof(aBuf), "starting to download map to '%s'", m_aMapdownloadFilenameTemp);
```
Comment (heinrich5991) [inline src/engine/client/client.cpp:1675]: It seems that this changes behavior, can you change it back?

**Review by heinrich5991 — COMMENTED** [review]
LGTM otherwise.

**Comment by heinrich5991** [comment]
bors r+

Thanks.

**Pull request merged**